File: Strivers A2Z DSA Sheet/1614-MaximumNestingDepthOfTheParentheses/1614-MaximumNestingDepthOfTheParentheses.java
// Last updated: 9/8/2026, 2:33:20 PM
class Solution {
    public int maxDepth(String s) {
        int p=0, ans=0;
        for(char c : s.toCharArray()){
            if(c == '(') p++;
            else if(c == ')') p--;

            ans = Math.max(ans, p);
        }
        return ans;
    }
}

File: Strivers A2Z DSA Sheet/1614-MaximumNestingDepthOfTheParentheses/1614-MaximumNestingDepthOfTheParentheses.cpp
// Last updated: 9/8/2026, 2:33:28 PM
class Solution {
public:
    int maxDepth(string s) {
        int p = 0;
        int ans = 0;
        for (char x : s) {
            // Increase depth on open parenthesis
            if (x == '(')
                p++;
            // Decrease depth on close parenthesis
            else if (x == ')')
                p--;
            // Update maximum depth encountered
            ans = max(ans, p);
        }
        return ans;
    }
};